src/components/SearchApp/SearchApp.js

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react'
+import classes from './Container.css'
+import Search from '../Search/Search'
+import Table from '../Table/Table'
+import CheckPrevious from '../CheckPrevious/CheckPrevious'
+
+const SearchApp = (props) => {
+  const [persons, setPersons] = useState(props.persons)
+  const [name, setName] = useState('')
+  const [pastSearches, setPastSearches] = useState([])
+
+  const handleChangeName = (event) => {
+    setName(event.target.value)
+  }
+
+  const getPersons = (name) => {
+    return props.persons.filter((person) => {
+      if (person.name.toLowerCase().startsWith(name)) {
+        return person
+      }
+      return null
+    })
+  }
+
+  const handleCheckboxSearch = (selectedItems) => {
+    if (selectedItems.length > 0) {
+      let tempPersons = selectedItems.map((selectedSearch) => {
+        return getPersons(selectedSearch)
+      })
+      setPersons(tempPersons.flat())
+    } else {
+      setPersons(props.persons)
+    }
+  }
+
+  const handleSearchName = (name) => {
+    const search = name.toLowerCase().trimStart().trimEnd()
+    //filters the person names list
+    const updatedPersons = getPersons(search)
```

Comment:
Function name should be Generic like updateItem. As we can use the same component for other type of items than person